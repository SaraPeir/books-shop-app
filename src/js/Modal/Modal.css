img {
    width: 70%;
   box-shadow: 10px 10px 10px lightgrey;
}

.centering {
    text-align: center;
}

.special-typo {
    font-weight: bold;
    font-style: italic;
}

.special-typo2 {
    font-weight: 100;
}