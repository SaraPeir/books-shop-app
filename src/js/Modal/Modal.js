import React, { Component } from 'react';
import './Modal.css';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

class ModalBS extends Component {

  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    const {modal} = this.state;
    this.setState({
      modal: !modal
    });
  }

  render() {
    return (
    <div>
      <div className="centering">
        <Button className="button" onClick={this.toggle}>Más detalles</Button>
      </div>
      
      <Modal onClick={this.toggle} isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
          <ModalHeader toggle={this.toggle}><span className={'special-typo2'}>{this.props.author},</span> <span className={'special-typo'}>{this.props.title}</span></ModalHeader>
          <ModalBody onClick={this.toggle} className={'container centering'}>
          <div className={'row'}>
            <div className={'col-12'}>
              <img className="img-style" src={this.props.url} alt={`${this.props.title} image`} />
            </div>
            <div className={'col-12  p-5'}>
              <p className={'special-typo'}>{this.props.introductionText}</p>
              <p>{this.props.content}</p>
            </div>
          </div>
        </ModalBody>
        <ModalFooter toggle={this.toggle}>{this.props.type2} - {this.props.pageNumber} páginas</ModalFooter>
      </Modal>
    </div>  
    );
  }
}

export default ModalBS;