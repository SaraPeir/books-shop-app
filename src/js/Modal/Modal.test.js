import React from "react";
import Modal from "./Modal";

import { shallow, mount } from 'enzyme';

describe('<Modal />', () => {

        it('should render without crashing', () => {
            const wrapper = shallow(<Modal />);
            expect(wrapper).toHaveLength(1);
        });

        it('should match snapshot', () => {
            const wrapper = shallow(<Modal />);
            expect(wrapper).toMatchSnapshot();
        });

        it('should show modal', () => {
            const mockState = {
                modal: false
            }

            const newState = {
                modal: true
            }
            
            const wrapper = shallow(<Modal />);
            wrapper.setState({ ...mockState })
            const newWrapper = wrapper.find('.button');
            newWrapper.simulate('click');
            expect(wrapper.instance().state).toEqual(newState);
        });
    });