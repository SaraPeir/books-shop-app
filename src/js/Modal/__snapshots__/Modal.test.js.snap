// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should match snapshot 1`] = `
<div>
  <div
    className="centering"
  >
    <Button
      className="button"
      color="secondary"
      onClick={[Function]}
      tag="button"
    >
      Más detalles
    </Button>
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClick={[Function]}
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="special-typo2"
      >
        ,
      </span>
       
      <span
        className="special-typo"
      />
    </ModalHeader>
    <ModalBody
      className="container centering"
      onClick={[Function]}
      tag="div"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <img
            alt="undefined image"
            className="img-style"
          />
        </div>
        <div
          className="col-12  p-5"
        >
          <p
            className="special-typo"
          />
          <p />
        </div>
      </div>
    </ModalBody>
    <ModalFooter
      tag="div"
      toggle={[Function]}
    >
       - 
       páginas
    </ModalFooter>
  </Modal>
</div>
`;
